### PR TITLE
Cleanup ListView and friends

### PR DESCRIPTION
**Pull request by hughbe**
Question: can we rename the fields in `ListViewItem` and `ListViewItem.ListViewSubItem` to match guidelines? Not sure as this type is serializable

### CODE REVIEW

**Comment by zsd4yr** [comment]
thanks for separating this into multiple commits.

One little nit-pick: it would be nice if the first one had *no* cleanup so that a review can just glance and see the lines added and lines subtracted are equal

**Comment by zsd4yr** [comment]
> Question: can we rename the fields in `ListViewItem` and `ListViewItem.ListViewSubItem` to match guidelines? Not sure as this type is serializable

@dreddy-work and @Tanya-Solyanik can talk about serialization and changes there. 

What guidelines? What would do they say we should change the names to?

**Comment by danmosemsft** [comment]
> @dreddy-work and @Tanya-Solyanik can talk about serialization and changes there.

BTW, what is your test strategy for binary serialization compatibility? In CoreFX, we cut down types marked `[Serializable]` to the minimum, because of the maintenance burden and for those we added tests that ensure we can serialize-deserialize between .NET Core and itself, and between .NET Core and .NET Framework. Writing the latter tests was pretty tricky though -we have to embed serialization payloads in the test.

1. Have you guys audited all your `[Serializable]` types to make sure they all need to be? Eg., they end up in resx files.
2. Do you need to add tests that ensure that you can exchange payloads with .NET Framework? For example in ListView, if `info.AddValue("Text", Text);` changed to use a different name then this would break. If you need to consume .NET Framework written .resx files, I assume this is important. @ericstj (Of cousre whether you have time to write such tests now is a different thing)

**Comment by hughbe** [comment]
> What guidelines? What would do they say we should change the names to?

That private fields should be prefixed with a `_`, e.g. `_privateField`

**Comment by zsd4yr** [comment]
> * Have you guys audited all your `[Serializable]` types to make sure they all need to be? Eg., they end up in resx files.

I do not believe we have done this.

> * Do you need to add tests that ensure that you can exchange payloads with .NET Framework? For example in ListView, if `info.AddValue("Text", Text);` changed to use a different name then this would break. If you need to consume .NET Framework written .resx files, I assume this is important. @ericstj (Of course whether you have time to write such tests now is a different thing)

I do not believe we have written such tests. I agree there is a need for them 😄

**Comment by hughbe** [comment]
I will not rename the fields! Is there an issue open in terms of auditing serializable?

**Comment by danmosemsft** [comment]
Serialization decisions tracked by https://github.com/dotnet/winforms/issues/890 and https://github.com/dotnet/winforms/issues/889

File: src/System.Windows.Forms/src/System/Windows/Forms/ListViewGroup.cs [hunk 1]
```diff
@@ -3,280 +3,207 @@
...
-
-        /// <include file='doc\ListViewGroup.uex' path='docs/doc[@for="ListViewGroup.ListViewGroup"]/*' />
-        /// <devdoc>
-        ///     Creates a ListViewGroup.
-        /// </devdoc>
-        public ListViewGroup() : this(string.Format(SR.ListViewGroupDefaultHeader, nextHeader++))
+    /// <summary>
+    /// Represents a group within a ListView.
+    /// </summary>
+    [TypeConverterAttribute(typeof(ListViewGroupConverter))]
```
Comment (danmosemsft) [inline src/System.Windows.Forms/src/System/Windows/Forms/ListViewGroup.cs]: I prefer this style of separate attributes,  but I guess the codebase should consistently use one or the other so the others would need cleanup later.

File: src/System.Windows.Forms/src/System/Windows/Forms/ListViewGroupCollection.cs [hunk 2]
```diff
@@ -9,349 +9,305 @@
...
+                if (_list == null)
+                {
                     return null;
                 }
 
-                for (int i = 0; i < list.Count; i ++) {
-                    if (string.Compare(key, this[i].Name, false /*case insensitive*/, System.Globalization.CultureInfo.CurrentCulture) == 0) {
+                for (int i = 0; i < _list.Count; i++)
+                {
+                    if (string.Equals(key, this[i].Name, StringComparison.CurrentCulture))
```
Comment (danmosemsft) [inline src/System.Windows.Forms/src/System/Windows/Forms/ListViewGroupCollection.cs]: I see the prevoius comment "case insensitive" was incorrect, so what you have here is correct.

File: src/System.Windows.Forms/src/System/Windows/Forms/ListViewGroupCollection.cs [hunk 3]
```diff
@@ -9,349 +9,305 @@
...
 
-        /// <include file='doc\ListViewGroup.uex' path='docs/doc[@for="ListViewGroup.ListViewGroupCollection.IList.Contains"]/*' />
-        /// <internalonly/>
+        public bool Contains(ListViewGroup value) => List.Contains(value);
+
         bool IList.Contains(object value)
         {
-            if (value is ListViewGroup) {
-                return Contains((ListViewGroup)value);
+            if (!(value is ListViewGroup group))
```
Comment (danmosemsft) [inline src/System.Windows.Forms/src/System/Windows/Forms/ListViewGroupCollection.cs]: some of these could reasonably be written on one line I think, up to you.
```c#
bool IList.Contains(object value) => (value is ListViewGroup group) ? Contains(group) : false;

File: src/System.Windows.Forms/src/System/Windows/Forms/ListViewGroupCollection.cs [hunk 4]
```diff
@@ -9,349 +9,305 @@
...
+        private void MoveGroupItems(ListViewGroup group)
+        {
+            Debug.Assert(_listView.IsHandleCreated, "MoveGroupItems pre-condition: listView handle must be created");
 
-            foreach(ListViewItem item in group.Items) {
-                if (item.ListView == this.listView) {
-                    item.UpdateStateToListView(item.Index);                    
+            foreach (ListViewItem item in group.Items)
+            {
+                if (item.ListView == this._listView)
```
Comment (danmosemsft) [inline src/System.Windows.Forms/src/System/Windows/Forms/ListViewGroupCollection.cs]: are you removing `this` ?

File: src/System.Windows.Forms/src/System/Windows/Forms/ListViewGroupCollection.cs (hunk 3 above)
Comment (danmosemsft) [reply src/System.Windows.Forms/src/System/Windows/Forms/ListViewGroupCollection.cs]: or
```c#
bool IList.Contains(object value) => (value is ListViewGroup group) && Contains(group);
```

File: src/System.Windows.Forms/src/System/Windows/Forms/ListViewGroupItemCollection.cs [hunk 5]
```diff
@@ -2,153 +2,107 @@
...
+        public ListViewItem Insert(int index, ListViewItem item)
         {
-            return Items.IndexOf(item);
-        }
-        
-        public ListViewItem Insert(int index, ListViewItem item) {            
             CheckListViewItem(item);
 
-            MoveToGroup(item, this.group);                        
+            MoveToGroup(item, this._group);
```
Comment (danmosemsft) [inline src/System.Windows.Forms/src/System/Windows/Forms/ListViewGroupItemCollection.cs]: more `this`. is there a rule?

File: src/System.Windows.Forms/src/System/Windows/Forms/ListViewItem.cs [hunk 6]
```diff
@@ -521,200 +522,208 @@ public ListViewItem(ListViewSubItem[] subItems, string imageKey) : this()
...
-            set {
-                if (value.Equals(position))
-                    return;
-                position = value;
-                if (listView != null && listView.IsHandleCreated) {
-                    if (!listView.VirtualMode) {
-                        listView.SetItemPosition(Index, position.X, position.Y);
+            set
+            {
+                if (!value.Equals(position))
```
Comment (danmosemsft) [inline src/System.Windows.Forms/src/System/Windows/Forms/ListViewItem.cs]: No need to change, but in preference I like early return as it avoids indenting.

**Comment by danmosemsft** [comment]
Merging as feedback is trivial and I don't want to have to review this again - phew. Thanks!

**Pull request merged**